lib/terraform/runner.rb

```diff
@@ -0,0 +1,293 @@
+      #        terraform-runner run
+      # @return [Terraform::Runner::ResponseAsync] Response object of terraform-runner create action
+      def run_async(input_vars, template_path, tags: nil, credentials: [], env_vars: {})
+        _log.info("In run_aysnc with #{template_path}")
+        response = create_stack_job(
+          template_path,
+          :input_vars  => input_vars,
+          :tags        => tags,
+          :credentials => credentials,
+          :env_vars    => env_vars
+        )
+        Terraform::Runner::ResponseAsync.new(response.stack_id)
+      end
+
+      # Stop running terraform-runner job by stack_id
+      #
+      # @param stack_id [String] stack_id from the terraforn-runner job
+      #
+      # @return [Terraform::Runner::Response] Response object with result of terraform run
+      def stop_async(stack_id)
+        cancel_stack_job(stack_id)
+      end
+
+      # Runs a template, waits until it terraform-runner job completes, via terraform-runner api
+      #
+      # @param input_vars [Hash] Hash with key/value pairs that will be passed as input variables to the
+      #        terraform-runner run
+      # @param template_path [String] Path to the template we will want to run
+      # @param tags [Hash] Hash with key/values pairs that will be passed as tags to the terraform-runner run
+      # @param credentials [Array] List of Authentication object ids to provide to the terraform run
+      # @param env_vars [Hash] Hash with key/value pairs that will be passed as environment variables to the
+      #        terraform-runner run
+      # @return [Terraform::Runner::Response] Response object with final result of terraform run
+      def run(input_vars, template_path, tags: nil, credentials: [], env_vars: {})
+        _log.info("Run template: #{template_path}")
+        create_stack_job_and_wait_until_completes(
+          template_path,
+          :input_vars  => input_vars,
+          :tags        => tags,
+          :credentials => credentials,
+          :env_vars    => env_vars
+        )
+      end
+
+      # Fetch terraform-runner job result/status by stack_id
+      #
+      # @param stack_id [String] stack_id from the terraforn-runner job
+      #
+      # @return [Terraform::Runner::Response] Response object with result of terraform run
+      def fetch_result_by_stack_id(stack_id)
+        retrieve_stack_job(stack_id)
+      end
+
+      # =================================================
+      # TerraformRunner Stack-API interaction methods
+      # =================================================
+      private
+
+      def server_url
+        ENV['TERRAFORM_RUNNER_URL'] || 'https://localhost:27000'
+      end
+
+      def server_token
+        # TODO: fix hardcoded token
+        ENV['TERRAFORM_RUNNER_TOKEN'] || 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IlNodWJoYW5naSBTaW5naCIsImlhdCI6MTcwNjAwMDk0M30.46mL8RRxfHI4yveZ2wTsHyF7s2BAiU84aruHBoz2JRQ'
```

Comment:
Maybe drop this hardcoded value and set it in your local environment for now?  We can also stub/mock it in tests if needed.  We don't want to accidentally post tokens.

Reply:
move hardcode value into test